Functions/Date/CountMonths.pq

```diff
@@ -0,0 +1,95 @@
+let pq_custom_function =
+
+let
+    // Define metadata for the function, describing its purpose and usage.
+    metaDocumentation = type function (
+        StartDate as (type date meta [
+            Documentation.FieldCaption = "Start Date",
+            Documentation.SampleValues = {" 01/01/2024 (dd/MM/yyyy) "},
+            // Documentation.AllowedValues = {"01/01/2024"},
+            Formatting.IsMultiLine = false,
+            Formatting.IsCode = false
+        ]),
+        EndDate as (type date meta [
+            Documentation.FieldCaption = "End Date",
+            Documentation.SampleValues = {" 31/01/2024 (dd/MM/yyyy) "},
+            // Documentation.AllowedValues = {"31/01/2025"},
+            Formatting.IsMultiLine = false,
+            Formatting.IsCode = false
+        ]),
+        optional CompletedMonthsOnly as (type nullable logical meta [
+            Documentation.FieldCaption = "Do you want to count completed months only?",
+            Documentation.FieldDescription = "Count by completed months (true) or actual months (false)",
+            Documentation.SampleValues = {"True", "False"},
+            Documentation.AllowedValues = {true,false},
+            Formatting.IsMultiLine = false,
+            Formatting.IsCode = true
+        ])
+    ) as list meta [
+        Documentation.Name = "fnCountMonths",
+        
+        Documentation.LongDescription = 
+        "    
+                <p><b>  Count Months Between Dates  </b></p>
+
+                <li>------------------------------------------------------</li>
+                
+                <li><b>  Creator: </b> Imran Haq  </li>
+                <li><b>  Web: </b> https://github.com/PBIQueryous  </li>
+                <li><b>  Acknowledgements: </b>   </li>
+                <li><b>  LinkedIn: </b>   </li>
+
+                <li>------------------------------------------------------</li>
+
+                <li><b>  Editor: </b>  Oscar Martinez  </li>
+                <li><b>  Web: </b>  https://bibb.pro  </li>
+                <li><b>  LinkedIn: </b>  https://www.linkedin.com/in/oscarmartinezv/  </li>
+
+                <li>------------------------------------------------------</li>
+
+                <p><b>  Function Description:  </b></br>
+                This function counts the number of months between the a selected Start Date and End Date. The third parameter allow you to count either complete months or actual months
+                
+                <p><b>  Parameters:  </b></br>
+                <ul>
+                    <li><b>  • StartDate:  </b>  The Start Date field from your data. <code> (eg: 01/01/2024) </code> </li>
+                    <li><b>  • EndDate:  </b>  The End Date field from your data. <code> (eg: 31/12/2025) </code> </li>
+                    <li><b>  • CompletedMonthsOnly:  </b>  Select 'true' to count only completed months, 'false' for actual months. </li>
```

Comment:
It would be useful to define what "completed month" means.

Is Jan-15-2023 to Feb-15-2023 a completed month? How about Jan-31-2023 to Feb-28-2023?

Reply:
In this context, a completed month is any entire month in the past, not including the current month.

Eg: Today is 26th April 2024. Let's say we want to count the months between Jan and April. That means there are 3 completed months, because April hasn't finished yet. Once we reach 1st May, then we will have 4 completed months, including April.